File: Python/Letter Pairs/Letter Pairs.py
with open("letters.txt") as letters:
    string: str = letters.read()

new = [string[0]]

for char in string[1:]:
    new.pop() if new and abs(ord(char) - ord(new[-1])) == 32 else new.append(char)

print("".join(new))
print(len(new))

File: Python/Letter Pairs/LetterPairs.java
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.util.Scanner;

public class LetterPairs {
    public static void main(String[] args) throws FileNotFoundException {
        Scanner scanner = new Scanner(new FileReader("letters.txt"));
        scanner.useDelimiter("");

        StringBuilder stringBuilder = new StringBuilder(scanner.next());

        while (scanner.hasNext()) {
            char next = scanner.next().charAt(0);
            if ((Math.abs(stringBuilder.charAt(stringBuilder.length() - 1) - next)) == 32) {
                stringBuilder.deleteCharAt(stringBuilder.length() - 1);
            } else {
                stringBuilder.append(next);
            }
        }

        System.out.println(stringBuilder);

        scanner.close();
    }
}
